File: python/server/routes/me.py
from fastapi import APIRouter, Depends, HTTPException, Request
from sqlalchemy.ext.asyncio import AsyncSession
from sqlalchemy import select

from ..database import get_db
from ..models.user import User
from ..utils import decode_token

router = APIRouter(prefix="/api/me")


@router.post("/status_broadcast")
async def toggle_status_broadcast(request: Request, session: AsyncSession = Depends(get_db)):
    auth = request.headers.get("authorization", "")
    payload = decode_token(auth)
    if not payload:
        raise HTTPException(status_code=401, detail="Unauthorized")
    user_id = int(payload["sub"])
    result = await session.execute(select(User).where(User.id == user_id))
    user = result.scalar_one_or_none()
    if not user:
        raise HTTPException(status_code=404, detail="User not found")
    body = await request.json()
    user.privacy_mode = body.get("privacy_mode", user.privacy_mode)
    await session.commit()
    return {"success": True}

File: js/apps/server/src/routes/me.ts
import { Router } from 'express';
import { prisma } from '../utils/database.js';
import { asyncHandler } from '../utils/http.js';
import { getUserIdFromAuthHeader } from '../services/authService.js';

export const meRouter = Router();

meRouter.post(
  '/status_broadcast',
  asyncHandler(async (req, res) => {
    const userId = await getUserIdFromAuthHeader(req.headers.authorization);
    const user = await prisma.user.findUnique({ where: { id: userId } });
    if (!user) {
      res.status(404).json({ detail: 'User not found' });
      return;
    }

    const { privacyMode } = (req.body ?? {}) as { privacyMode?: string };
    if (privacyMode !== undefined) {
      if (!['friends', 'none', 'everyone'].includes(privacyMode)) {
        res.status(400).json({ detail: 'Invalid privacy_mode value' });
        return;
      }
      await prisma.user.update({ where: { id: userId }, data: { privacyMode } });
    }

    res.json({ success: true });
  })
);
